RuntimeError: Event loop is closed

After upgrading to `pytest==5.4.3 pytest-asyncio==0.14.0` I'm getting transient(?) test failures and all I see in the log is this:

```py
=================================== FAILURES ===================================
___________________________ test_xx ___________________________

kwargs = {'xx': XxFixture(, ...}
 
    @functools.wraps(func)
    def inner(**kwargs):
>       coro = func(**kwargs)
 
/root/.cache/pypoetry/virtualenvs/xx-py3.8/lib/python3.8/site-packages/pytest_asyncio/plugin.py:175: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/root/.cache/pypoetry/virtualenvs/xx-py3.8/lib/python3.8/site-packages/pytest_asyncio/plugin.py:175: in inner
    coro = func(**kwargs)
/root/.cache/pypoetry/virtualenvs/xx-py3.8/lib/python3.8/site-packages/pytest_asyncio/plugin.py:177: in inner
    task = asyncio.ensure_future(coro, loop=_loop)
/usr/local/lib/python3.8/asyncio/tasks.py:661: in ensure_future
    task = loop.create_task(coro_or_future)
/usr/local/lib/python3.8/asyncio/base_events.py:429: in create_task
    self._check_closed()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <_UnixSelectorEventLoop running=False closed=True debug=False>

    def _check_closed(self):
        if self._closed:
>           raise RuntimeError('Event loop is closed')
E           RuntimeError: Event loop is closed

/usr/local/lib/python3.8/asyncio/base_events.py:508: RuntimeError
---------------------------- Captured stdout setup -----------------------------
```

There are some fragile tests in the project and we use flaky to manage those, so I see this error only when some test fails thrice.
Weirdly I don't get to see the actual test failure/cause in this case.

At the moment I'm at a loss as to how to debug it 🤔

--- a/.gitignore
+++ b/.gitignore
@@ -59,3 +59,4 @@ target/
 
 .venv*
 .idea
+.vscode

--- a/pytest_asyncio/plugin.py
+++ b/pytest_asyncio/plugin.py
@@ -186,6 +186,12 @@ def wrap_in_sync(func, _loop):
     """Return a sync wrapper around an async function executing it in the
     current event loop."""
 
+    # if the function is already wrapped, we rewrap using the original one
+    # not using __wrapped__ because the original function may already be
+    # a wrapped one
+    if hasattr(func, '_raw_test_func'):
+        func = func._raw_test_func
+
     @functools.wraps(func)
     def inner(**kwargs):
         coro = func(**kwargs)
@@ -201,6 +207,7 @@ def wrap_in_sync(func, _loop):
                     task.exception()
                 raise
 
+    inner._raw_test_func = func
     return inner
 
 

--- a/setup.py
+++ b/setup.py
@@ -47,6 +47,7 @@ setup(
         "testing": [
             "coverage",
             "hypothesis >= 5.7.1",
+            "flaky >= 3.5.0"
         ],
     },
     entry_points={"pytest11": ["asyncio = pytest_asyncio.plugin"]},

--- a/tests/test_flaky_integration.py
+++ b/tests/test_flaky_integration.py
@@ -1,0 +1,17 @@
+"""Tests for the Flaky integration, which retries failed tests.
+"""
+import asyncio
+
+import flaky
+import pytest
+
+_threshold = -1
+
+
+@flaky.flaky(3, 2)
+@pytest.mark.asyncio
+async def test_asyncio_flaky_thing_that_fails_then_succeeds():
+    global _threshold
+    await asyncio.sleep(0.1)
+    _threshold += 1
+    assert _threshold != 1

